### DISCUSSION TITLE
How to safely clean up old dump files for freeing space?

### DISCUSSION THREAD

**Discussion by bzon** [dragonflydb/dragonfly#1566] [Q&A] [answered]
I have run dragonfly with `--save_schedule:*:10` (every 10 minutes). I noticed that it keeps a lot of snapshots by default. Is it safe to delete these files? If I will create a clean up script, should I just keep the latest snapshots (by timestamp)? Is 

```
-rw-r--r-- 1 root 1001 323M Jul 18 15:10 dump-2023-07-18T15:10:17-0000.dfs
-rw-r--r-- 1 root 1001 322M Jul 18 15:10 dump-2023-07-18T15:10:17-0003.dfs
-rw-r--r-- 1 root 1001 322M Jul 18 15:10 dump-2023-07-18T15:10:17-0002.dfs
-rw-r--r-- 1 root 1001 322M Jul 18 15:10 dump-2023-07-18T15:10:17-0006.dfs
-rw-r--r-- 1 root 1001 322M Jul 18 15:10 dump-2023-07-18T15:10:17-0004.dfs
-rw-r--r-- 1 root 1001 322M Jul 18 15:10 dump-2023-07-18T15:10:17-0001.dfs
-rw-r--r-- 1 root 1001 322M Jul 18 15:10 dump-2023-07-18T15:10:17-0005.dfs
-rw-r--r-- 1 root 1001 4.0K Jul 18 15:10 dump-2023-07-18T15:10:17-summary.dfs
-rw-r--r-- 1 root 1001 336M Jul 18 16:10 dump-2023-07-18T16:10:04-0000.dfs
-rw-r--r-- 1 root 1001 335M Jul 18 16:10 dump-2023-07-18T16:10:04-0005.dfs
-rw-r--r-- 1 root 1001 335M Jul 18 16:10 dump-2023-07-18T16:10:04-0002.dfs
-rw-r--r-- 1 root 1001 335M Jul 18 16:10 dump-2023-07-18T16:10:04-0006.dfs
-rw-r--r-- 1 root 1001 335M Jul 18 16:10 dump-2023-07-18T16:10:04-0004.dfs
-rw-r--r-- 1 root 1001 335M Jul 18 16:10 dump-2023-07-18T16:10:04-0001.dfs
-rw-r--r-- 1 root 1001 4.0K Jul 18 16:10 dump-2023-07-18T16:10:04-summary.dfs
-rw-r--r-- 1 root 1001 335M Jul 18 16:10 dump-2023-07-18T16:10:04-0003.dfs
```

**Comment by romange** [comment, marked as answer]
hi @bzon . As explained by https://www.dragonflydb.io/docs/managing-dragonfly/backups our default mode is using `dump-{timestamp}` filenames. However, if you want you can override the flag and just use `dump` which will override the same snapshot again and again.

**Reply by bzon** [reply]
If I change the flag value for `dbfilename`  now of a running dragonfly, I need to restart it right? Will it use the old dump files? I'm worried that it will restart and will not restore the old snapshot and lose data.

**Reply by romange** [reply]
i can confirm it will still load the data from the latest snapshot.

**Reply by admtech** [reply]
The idea of {timestamp} is good in itself, but it would be nice if you could refine it a bit. For example, I would like to save the database for each day, so rather the value {date} or even finer {Y}-{m}-{d} (so that you can set it e.g. only daily, monthly or yearly). With {timestamp} the system fills up very fast if you use the "save" function often or if you time it. But if you only have the current date, you would always have a backup of the previous days etc. automatically. The rest, i.e. the cleanup, can be done by a shell script (delete everything older than x days etc.). 

I have opened a feature issue for this:
https://github.com/dragonflydb/dragonfly/issues/1637

greetings
Frank

**Reply by cristianrat** [reply]
how do you customize this when running on k8s?
should be in the example if you ask me :)

**Reply by romange** [reply]
what example @cristianrat ?
we have a partial example here: https://www.dragonflydb.io/docs/managing-dragonfly/backups#the-dbfilename-flag
though I agree it's not enough

**Reply by cristianrat** [reply]
I had assumed it's possible via k8s to just specify the name in the snapshot section, but from what I see so far (please correct me if I am wrong) I need to pass it as an arg?
this is how my terraform looks now
```
.......
spec:
  replicas: ${local.min_replicas}
  image: docker.dragonflydb.io/dragonflydb/dragonfly:${local.dragonfly_version}
  args: ["--dbfilename=snapshot_ftw"]
  authentication:
...
```
In the docs (which are reasonably good) it would be a good idea to make a note: pruning is not automatic and you have to deal with that yourself
I was caught out by this and my disk filled up (yes, you can say my APM is not proper, and it wasn't :) clearly )

**Reply by romange** [reply]
first of all - thanks for the suggestion! And its good idea to add this comment - you are not the first one who bit by that. You are more than welcome to send a PR, btw - our docs are located there: 
https://github.com/dragonflydb/documentation - it can even account for hacktoberfest if you participate.

**Reply by romange** [reply]
you need to overwrite the dbfilename with a constant name like `--dbfilename=snapshot` to not have multiple snapshots. Unfortunately I do not know terraform to comment how it should be configured there.

**Reply by cristianrat** [reply]
already in prod, so that's the right way :) via terraform
if anybody wants the whole manifest, let me know, but should be clear enough
will raise a PR to the docs
